### PR TITLE
Connect Backend DCT Data to Front End / Serve Time Helper Method Implementation

### LINKED ISSUE #120: [5pts] (Front/Back)end: Connect DCT Data
**Fill in the blanks**:
As a user, I would like to see live DCT data, so that I can see accurate time periods and meals.

**Describe the solution you'd like:**
- [x] Create a `GET` request for a general "gimme all the data" route
- [x] On `GET` request, compile all of the DCT data into JSON and send it to the user
  - This data should include: meal times, stations, meals per station
- [x] Update the homepage and all-food menu to use this scrapped data

Note: You may need to move the mocked data temporarily over to the backend while the scraper is finalized

**Definition of Done - Check as you go!**
- [x] Tasks have been implemented
- [x] Documented
- [x] Passing Tests for each task
- [x] Pull Request made
- [x] Code Reviewed

### LINKED ISSUE #131: [2pts] Helper Method: Meal times & Meal names
**Fill in the blanks**:
As a developer, I would like to save Meal times and Meal names, so that it shares the workload of the web scraper task in-line with how we've been sharing the work thus far in the project.

**Describe the solution you'd like:**
- [x] A meaningful storage of meal times & meal names

**Definition of Done - Check as you go!**
- [x] Tasks have been implemented
- [x] Documented
- [x] Passing Tests for each task
- [x] Pull Request made
- [x] Code Reviewed

### PR DESCRIPTION
**Pull request by oreorz**
This pull request implements the connection from the backend server to the frontend in order to show the menu items on the home page. Additionally, I also created a helper method for saving the serving times at the DCT.

### CODE REVIEW

File: quack_app/lib/core/menu_data.dart [hunk 1]
```diff
@@ -16,6 +19,23 @@ class MenuData {
 
   MenuData._internal();
 
+  Future getDCTData() async {
```
Comment (RafaelPiloto10) [inline quack_app/lib/core/menu_data.dart]: Can we merge this code into the `loadData` method? If this method fails, what's the point in running the `loadData` function. If we merge it, then we just exit and don't have to worry about "key does not exist" errors.

File: quack_app/lib/screens/login/login_screen.dart [hunk 2]
```diff
@@ -13,6 +13,7 @@ class LoginScreen extends StatefulWidget {
 }
 
 class _LoginScreenState extends State<LoginScreen> {
+  late MenuData menuData;
```
Comment (RafaelPiloto10) [inline quack_app/lib/screens/login/login_screen.dart]: Don't instantiate it, use it as a Singleton: `MenuData().something`. That's actually what you're doing -- no need to define it.

File: dct_backend/server_test.py [hunk 3]
```diff
@@ -6,5 +6,14 @@
 def test_root():
     response = client.get("/")
     assert response.status_code == 200
-    assert response.json() == {"quack": "Hello World!"}
+    assert response.json() == {"msg": "This is the DCT Backend!"}
+
+
+data = {"stations": {"Stem To Root": {"name": "Stem To Root", "menu": [{"name": "Pancakes", "station": "Stem To Root", "meal_time": "Breakfast", "calories": "100", "categories": ["carbs"]}, {"name": "Bagel", "station": "Stem To Root", "meal_time": "Breakfast", "calories": "100", "categories": ["carbs"]}], "line_speed": 0}, "Hearth and Stone": {"name": "Hearth and Stone", "menu": [{"name": "Cereal", "station": "Hearth and Stone", "meal_time": "Lunch", "calories": "102", "categories": ["vegan", "carbs"]}], "line_speed": 0}, "605 Kitchen": {"name": "605 Kitchen", "menu": [{"name": "Sandwich", "station": "605 Kitchen", "meal_time": "Lunch", "calories": "102", "categories": ["category1"]}, {"name": "Avacado Grilled Cheese", "station": "605 Kitchen", "meal_time": "Dinner", "calories": "102", "categories": ["vegan", "carbs"]}], "line_speed": 0}, "Spice": {"name": "Spice", "menu": [
```
Comment (RafaelPiloto10) [inline dct_backend/server_test.py]: This must have been fun to make. Thanks for doing this!

**Review by RafaelPiloto10 — CHANGES_REQUESTED** [review]
Huge core functionality update! Thank you for doing this! Can't wait to see things populate with real data. Some minor tweaking to how the app loads the data, but everything else looks stellar!

File: quack_app/lib/core/menu_data.dart (hunk 1 above)
Comment (oreorz) [reply quack_app/lib/core/menu_data.dart]: You're right, this is one of the fixes I was looking for. I'll do this

**Comment by RafaelPiloto10** [comment]
@oreorz Here's something to note: Your flutter tests will fail no matter what because the `getLoad()` function won't be able to connect to the backend at `127.0.0.1`. You can take 2 approaches: 1. wait for me to upload the backend and just attach a screenshot of all tests passing on your machine. 2. Mock the `getLoad()` function in tests and set the data to mocked/test data.

Which ever you prefer. Ideally, tests shouldn't be making GET requests and we should be mocking everything. But given time constraints, I wouldn't be mad if you just wanted to wait until we deploy the backends

**Review by RafaelPiloto10 — APPROVED** [review]
Kuddos for this one! You deserve a real life taco for having to write those test dictionaries manually

File: dct_backend/server_test.py [hunk 4]
```diff
@@ -6,5 +6,15 @@
...
+    assert response.json() == {"msg": "This is the DCT Backend!"}
+
+
+data = {"stations": {"Stem To Root": {"name": "Stem To Root", "menu": [{"name": "Pancakes", "station": "Stem To Root", "meal_time": "Breakfast", "calories": "100", "categories": ["carbs"]}, {"name": "Wings", "station": "Stem To Root", "meal_time": "Lunch", "calories": "102", "categories": ["category1"]}, {"name": "Macaroni & Cheese", "station": "Stem To Root", "meal_time": "Dinner", "calories": "102", "categories": ["vegan", "carbs"]}, {"name": "Pasta", "station": "Stem To Root", "meal_time": "Breakfast", "calories": "100", "categories": ["carbs"]}], "line_speed": 0}, "605 Kitchen": {"name": "605 Kitchen", "menu": [{"name": "Bagel", "station": "605 Kitchen", "meal_time": "Lunch", "calories": "100", "categories": ["carbs"]}, {"name": "Grilled Chicken", "station": "605 Kitchen", "meal_time": "Dinner", "calories": "102", "categories": ["vegan", "carbs"]}, {"name": "Veggie Fried Rice", "station": "605 Kitchen", "meal_time": "Breakfast", "calories": "100", "categories": ["carbs"]}, {"name": "French Fries", "station": "605 Kitchen", "meal_time": "Lunch", "calories": "100", "categories": [
```
Comment (Foxworth22) [inline dct_backend/server_test.py:12]: These seem like lengthy tests. 
- How did you make them?
- This isn't hard-coded menu info, right?

File: quack_app/lib/core/menu_data.dart [hunk 5]
```diff
@@ -24,46 +29,25 @@ class MenuData {
...
-      "Breakfast",
-      "Lunch",
-      "Dinner"
-    ]; // Serve times will be parsed here from backend
-    _stations = [
-      "Stem To Root",
-      "605 Kitchen",
-      "Spice",
-      "Luncheonnette"
-    ]; // Stations will be parsed here from backend
```
Comment (Foxworth22) [inline quack_app/lib/core/menu_data.dart]: Why was this comment removed?

**Review by Foxworth22 — APPROVED** [review]
Looks good! Going forward I would like to see more in-line comments. Pretend you haven't looked at the code in over a year and have to explain it to yourself.

File: quack_app/lib/core/menu_data.dart (hunk 5 above)
Comment (RafaelPiloto10) [reply quack_app/lib/core/menu_data.dart]: This comment is a reminder for what this PR implements

File: dct_backend/server_test.py (hunk 4 above)
Comment (RafaelPiloto10) [reply dct_backend/server_test.py:12]: This is our mocked menu data put into explicit JSON format. So your webscraper should ideally return something similar to this. 

For tests, this is hardcoded

**Pull request merged**